### ISSUE TITLE
Eureka discovery monitoring not working, when instance name is not equals with app name

### ISSUE THREAD

**Issue by szabolcsgnr** [codecentric/spring-boot-admin#552]
`de.codecentric.boot.admin.discovery.ApplicationDiscoveryListener#discover` calls `org.springframework.cloud.client.discovery.DiscoveryClient#getInstances` with the retrieved Application IDs.

When you are using eureka for service registration you can supply an instance ID like that:
```yaml
eureka:
  instance:
    appName: app-name
```

If are using the configuration bootstrap first approach of spring cloud you should supply an app name as well.
```yaml
spring:
  application:
    name: other-app-name-same-app
```

Unfortunatly, eureka discovery client tries to return the instances by the vip address, which in this case will be other-app-name-same-app. As the method invoked with app-name, it will return an empty list, causing these services won`t be registered in the admin application.

In my opinion, it is contract violation in eureka client implementation. Has someone other came across with that issue? I would be open to any suggestions.

**Comment by joshiste** [comment]
Spring Boot Admin uses the Spring Cloud's `DiscoveryClient` interface. This interface defines two methods:
```java
	/**
	 * Get all ServiceInstances associated with a particular serviceId
	 * @param serviceId the serviceId to query
	 * @return a List of ServiceInstance
	 */
	List<ServiceInstance> getInstances(String serviceId);

	/**
	 * @return all known service ids
	 */
	List<String> getServices();
```

The `ApplicationDiscoveryListener` uses the `getServices` to get all services and `getInstances`to retrieve all instances for each service. Which imho is totally compliant to the interface.
It doesn't know anything about eureka and how you registered your services.
Imho if you have problems when `spring.application.name` is not equal to `eureka.instance.appName` it's a spring-cloud-netflix-issue. (And also I don't get why you want two different names for the same application)

**Comment by szabolcsgnr** [comment]
Yes I wrote that this is a contract violation in the interface's implementation, and I agree that spring-boot-admin shouldn't be prepared for that issue. I had to use different name for the spring application and the eureka.instance.appName, as it was request from my peer.

If someone faces this issue, there's an easy solution: extend ApplicationDiscoveryListener and ovveride the discover() method. Using the EurekaClient all applications can be retrieved with it's instances, and there's no need to first query for the Application IDs, and the the individual instances. 

ApplicationDiscoveryListener bean initializaion is annotated as conditional, it can be easily replaced.
